chapter2/tests/t11/nginx.conf

```diff
@@ -0,0 +1,75 @@
+user nginx nginx;
+worker_processes 1;
+pid /run/nginx.pid;
+
+events {
+	worker_connections 768;
+}
+
+http {
+
+	types_hash_max_size 4096;
+	include /etc/nginx/mime.types;
+	default_type application/octet-stream;
+	ssl_protocols TLSv1 TLSv1.1 TLSv1.2 TLSv1.3;
+	ssl_prefer_server_ciphers on;
+	error_log /dev/stderr;
+	access_log /dev/stdout;
+
+
+	upstream php-fpm-slow {
+		server unix:/run/php/php8.1-fpm-slow.sock fail_timeout=0;
+	}
+
+	upstream php-fpm {
+		server unix:/run/php/php8.1-fpm.sock;
+	}
+
+	server {
+		server_name _;
+		listen 443 ssl;
+		ssl_certificate /tmp/certs/cert.pem;
+		ssl_certificate_key /tmp/certs/key.pem;
+		fastcgi_param  QUERY_STRING       $query_string;
+		fastcgi_param  REQUEST_METHOD     $request_method;
+		fastcgi_param  CONTENT_TYPE       $content_type;
+		fastcgi_param  CONTENT_LENGTH     $content_length;
+		fastcgi_param  SCRIPT_NAME        $fastcgi_script_name;
+		fastcgi_param  REQUEST_URI        $request_uri;
+		fastcgi_param  DOCUMENT_URI       $document_uri;
+		fastcgi_param  DOCUMENT_ROOT      $document_root;
+		fastcgi_param  SERVER_PROTOCOL    $server_protocol;
+		fastcgi_param  REQUEST_SCHEME     $scheme;
+		fastcgi_param  HTTPS              $https if_not_empty;
+
+		fastcgi_param  GATEWAY_INTERFACE  CGI/1.1;
+		fastcgi_param  SERVER_SOFTWARE    nginx/$nginx_version;
+
+		fastcgi_param  REMOTE_ADDR        $remote_addr;
+		fastcgi_param  REMOTE_PORT        $remote_port;
+		fastcgi_param  REMOTE_USER        $remote_user;
+		fastcgi_param  SERVER_ADDR        $server_addr;
+		fastcgi_param  SERVER_PORT        $server_port;
+		fastcgi_param  SERVER_NAME        $server_name;
+		fastcgi_param  SCRIPT_FILENAME    $realpath_root$fastcgi_script_name;
+		# PHP only, required if PHP was built with --enable-force-cgi-redirect
+		fastcgi_param  REDIRECT_STATUS    200;
+
+		location / {
+			fastcgi_split_path_info ^(.+\.php)(.*)$;
+		}
+
+		location /reports/fast/ {
+			root /tmp/camp/reports/fast;
+			fastcgi_pass php-fpm;
+			try_files /fast.php =404;
+		}
+
+		location /reports/slow/ {
+			root /tmp/camp/reports/slow;
+			fastcgi_pass php-fpm-slow;
+			fastcgi_read_timeout 605;
+			try_files /slow.php =404;
```

Comment:
define root at the server level, in your case it should be `/tmp/camp`
then have try_files as the first line in the location context. all `fastcgi_*` directives should be at the end of the location context.